NetCasbin/Enforcer.cs

```diff
@@ -79,177 +87,248 @@ public List<string> GetUsersForRoles(string[] names)
+            var parameters = new[] { user, role }.Concat(domain);
+            return RemoveGroupingPolicy(parameters.ToList());
         }
 
         /// <summary>
         /// Deletes a role for a user.
         /// </summary>
         /// <param name="user"></param>
         /// <param name="role"></param>
+        /// <param name="domain"></param>
         /// <returns>Returns false if the user does not have the role (aka not affected).</returns>
-        public Task<bool> DeleteRoleForUserAsync(string user, string role)
+        public Task<bool> DeleteRoleForUserAsync(string user, string role, params string[] domain)
         {
-            return RemoveGroupingPolicyAsync(user, role);
+            var parameters = new[] { user, role }.Concat(domain);
+            return RemoveGroupingPolicyAsync(parameters.ToList());
         }
 
         /// <summary>
         /// Deletes all roles for a user.
         /// </summary>
         /// <param name="user"></param>
+        /// <param name="domain"></param>
         /// <returns>Returns false if the user does not have any roles (aka not affected).</returns>
-        public bool DeleteRolesForUser(string user)
+        public bool DeleteRolesForUser(string user, params string[] domain)
         {
-            return RemoveFilteredGroupingPolicy(0, user);
+            string[] fieldValues = domain.Length switch
+            {
+                0 => new[] {user},
+                > 1 => throw new InvalidOperationException("Domain should be 1 parameter."),
+                _ => new[] {user, string.Empty, domain[0]}
+            };
+            return RemoveFilteredGroupingPolicy(0, fieldValues);
         }
 
         /// <summary>
         /// Deletes all roles for a user.
         /// </summary>
         /// <param name="user"></param>
+        /// <param name="domain"></param>
         /// <returns>Returns false if the user does not have any roles (aka not affected).</returns>
-        public Task<bool> DeleteRolesForUserAsync(string user)
+        public Task<bool> DeleteRolesForUserAsync(string user, params string[] domain)
         {
-            return RemoveFilteredGroupingPolicyAsync(0, user);
+            string[] fieldValues = domain.Length switch
+            {
+                0 => new[] {user},
+                > 1 => throw new InvalidOperationException("Domain should be 1 parameter."),
+                _ => new[] {user, string.Empty, domain[0]}
+            };
+            return RemoveFilteredGroupingPolicyAsync(0, fieldValues);
         }
 
         /// <summary>
-        /// DeleteUser deletes a user
+        /// Deletes a user
         /// </summary>
         /// <param name="user"></param>
         /// <returns>Returns false if the user does not exist (aka not affected).</returns>
         public bool DeleteUser(string user)
         {
-            return RemoveFilteredGroupingPolicy(0, user);
+            bool groupResult = RemoveFilteredGroupingPolicy(0, user);
```

Comment:
This doesn't relate to the domain thing right? I agree with this change.

Reply:
Yes, it just a little change.